testsuite/python/writevtf.py

```diff
@@ -43,64 +44,112 @@ class CommonTests(ut.TestCase):
     system.cell_system.skin = 0.4
     system.time_step = 0.01
     written_pos = None
+    written_bonds = None
+    written_atoms = None
+    
+
     types_to_write = None
     for i in range(npart):
         system.part.add(id=i, pos=np.array([float(i),
                                             float(i),
                                             float(i)]),
                         v=np.array([1.0, 2.0, 3.0]), type=1+(-1)**i)
+
+    system.bonded_inter.add(interactions.FeneBond(k=1., d_r_max=10.0))
+    system.part[0].add_bond((0,1))
+    system.part[0].add_bond((0,2))
+    system.part[0].add_bond((0,3))
+
     system.integrator.run(steps=0)
 
 
     def test_pos(self):
         """Test if positions have been written properly."""
         if self.types_to_write=='all': simulation_pos=np.array([((i), float(i), float(i), float(i)) for i in range(npart)]),
         elif (2 in self.types_to_write): simulation_pos=np.array([((i*2), float(i*2), float(i*2), float(i*2)) for i in range(npart//2)]),
+
         self.assertTrue(np.allclose(
             simulation_pos, self.written_pos),
             msg="Positions not written correctly by writevcf!")
 
 
+    def test_bonds(self):
+        """Test if bonds have been written properly."""
+        if self.types_to_write=='all': 
+            simulation_bonds=np.array([1,2,3]) #the two bonded particles 
+        elif (2 in self.types_to_write): 
+            types=[2]
+            simulation_bonds=np.array(2) # only this one is type 2
+        
+        self.assertTrue(np.allclose(
+            simulation_bonds, self.written_bonds),
+            msg="Bonds not written correctly by writevsf!")
+
+    def test_atoms(self):
+        """Test if atom declarations have been written properly."""
+        if self.types_to_write=='all': 
+            simulation_atoms=np.array([((i), (1+(-1)**i)) for i in range(npart)])
+        elif (2 in self.types_to_write): 
+            simulation_atoms=np.array([((i*2), 2) for i in range(npart//2)])
+
+        self.assertTrue(np.allclose(
+            simulation_atoms, self.written_atoms),
+            msg="Atoms not written correctly by writevsf!")
+
+
 class VCFTestAll(CommonTests):
     """
     Test the writing VTF files.
     """
     @classmethod
     def tearDownClass(cls):
-        os.remove("test.vtf")
+        os.remove("test.vcf")
```

Comment:
Please use [`tempfile`](https://docs.python.org/3/library/tempfile.html) rather than creating and removing files in the working directory.  This might be dangerous.